### PR TITLE
Performance improvement of SSLLayer::IsSSLMessage()

### PR DESCRIPTION
**Pull request by gx740**
1. Performance of SSLLayer::IsSSLMessage method was significantly improved
2. New SSL ports were added (I am not sure that it is necessary)

### CODE REVIEW

**Comment by gx740** [comment]
> In this PR you're replacing the port map with a simple switch case statement, but at the same time keeping the port map (which is now being used only in `SSLLayer::getSSLPortMap()`.

Unfortunately, `SSLLayer::getSSLPortMap()` had been made public so I could not simply remove this map.

> I think there's no point in this duplicate. Why is the switch case statement faster than the map? Is it somehow related to the benchmark you made in #290 ?

Operator `switch` will faster than the lookup in map (or sorted vector) any time:

1. Its complexity is constant
2. Generally, the compilers are optimized it very well

> Instead of keeping both map and switch case statement, what we can do is:
> 
> * Keep the map as is
> * Add this to optimize for most cases:
>   ```
>   if (port == 443) // HTTPS, this is likely case
>        return true;
>   ```

It's better to remove `SSLLayer::getSSLPortMap()` at all because it's useless for end user. Why?
- The SSL ports are well known and the new ones are rarely added
- Lookup in map (or in vector as described in #290) is very slow comparing to switch operator
- The logic of SSLLayer does not depend on certain port number

I've made the tests for the worst case (lookup of 995):
```
Measurement of lookup performance: container size is 14
Operator switch: duration(ms): 15, iterations: 1000000
Lookup in map: duration(ms): 1045, iterations: 1000000
```

Source code of tests:
```
#include <vector>
#include <map>
#include <cstdint>
#include <chrono>
#include <iostream>
#include <algorithm>

static std::map<uint16_t, bool> createSSLPortMap()
{
  std::map<uint16_t, bool> result;

  result[0] = true; //default
  result[261] = true; //NSIIOPS
  result[443] = true; //HTTPS
  result[448] = true; //DDM-SSL
  result[465] = true; //SMTPS
  result[563] = true; //NNTPS
  result[614] = true; //SSHELL
  result[636] = true; //LDAPS
  result[989] = true; //FTPS - data
  result[990] = true; //FTPS - control
  result[992] = true; //Telnet over TLS/SSL
  result[993] = true; //IMAPS
  result[994] = true; //IRCS
  result[995] = true; //POP3S

  return result;
}

static const std::map<uint16_t, bool> SSLPortMap = createSSLPortMap();

static bool isSSLPort(uint16_t port)
{
  if(port == 443) // HTTPS, this is likely case
    return true;

  switch(port)
  {
    case 0:   // default
    case 261: // NSIIOPS
    case 448: // DDM-SSL
    case 465: // SMTPS
    case 563: // NNTPS
    case 614: // SSHELL
    case 636: // LDAPS
    case 989: // FTPS - data
    case 990: // FTPS - control
    case 992: // Telnet over TLS/SSL
    case 993: // IMAPS
    case 994: // IRCS
    case 995: // POP3S
      return true;
    default:
      return false;
  }
}

int main(int argc, char *argv[])
{
  constexpr auto tryCount = 1000000U;
  constexpr uint16_t portToSearch = 995;
  auto count = 0U;

  std::cout << "Measurement of lookup performance: container size is " << SSLPortMap.size() << '\n';

  // function
  auto startTime = std::chrono::steady_clock::now();

  for(auto i = 0U; i < tryCount; ++i)
    if(isSSLPort(portToSearch))
      ++count;

  auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - startTime);
  std::cout << "Operator switch: duration(ms): " << duration.count() << ", iterations: " << count << '\n';

  // map
  count = 0;
  startTime = std::chrono::steady_clock::now();

  for(auto i = 0U; i < tryCount; ++i)
    if(SSLPortMap.find(portToSearch) != SSLPortMap.cend())
      ++count;

  duration = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - startTime);
  std::cout << "Lookup in map: duration(ms): " << duration.count() << ", iterations: " << count << '\n';

  return 0;
}
```

**Comment by gx740** [comment]
We can append a new public method `isSSLPort` instead of `getSSLPortMap`.

**Comment by seladb** [comment]
> We can append a new public method `isSSLPort` instead of `getSSLPortMap`

Yes, that makes sense to me. It'll break the current API but I don't think it's a major concern because as you mentioned not a lot of people use it. You will have to change the `SSLAnalyzer` example app though

**Comment by gx740** [comment]
The changes have been made

File: Packet++/header/SSLLayer.h [hunk 1]
```diff
@@ -249,7 +248,7 @@ namespace pcpp
 		 */
 		void parseNextLayer();
 
-        OsiModelLayer getOsiModelLayer() const { return OsiModelPresentationLayer; }
+    OsiModelLayer getOsiModelLayer() const { return OsiModelPresentationLayer; }
```
Comment (seladb) [inline Packet++/header/SSLLayer.h]: Please fix the indentation here

Comment (gx740) [reply Packet++/header/SSLLayer.h]: Done

**Pull request merged**